### PR TITLE
[Live-Site] feat: added api to kickout peer

### LINKED ISSUE #1374: [Live-Site] Feat/event kickout peer api
# Parent Story

https://github.com/Real-Dev-Squad/website-www/issues/343

# Description
Need an API to allow host to kick out peer from the event.

<html>
<body>

Method | Route | Description
-- | -- | --
PATCH | :id/peers/kickout | Removes peer from the event & update the database.

</body>
</html>

### PR DESCRIPTION
**Pull request by prerana1821**
# Issues

Fixes #1374 

# Added API:

- [x] PATCH  - /:id/peers/kickout - Removes peer from the event & update the database.

### CODE REVIEW

File: controllers/events.js [hunk 1]
```diff
@@ -226,6 +227,38 @@ const addPeerToEvent = async (req, res) => {
...
+  const payload = {
+    peer_id: req.body.peerId,
+    reason: req.body.reason,
+  };
+
+  try {
+    await apiService.post(`/active-rooms/${id}/remove-peers`, payload);
+    await eventQuery.kickoutPeer({ eventId: id, peerId: payload.peer_id, reason: req.body.reason });
+    return res.status(200).json({
+      message: `Peer is kicked out from the event`,
```
Comment (iamitprakash) [inline controllers/events.js:251]: Selected Participant is removed from event

File: controllers/events.js [hunk 2]
```diff
@@ -226,6 +227,38 @@ const addPeerToEvent = async (req, res) => {
...
+    });
+  } catch (error) {
+    logger.error({ error });
+    return res.status(500).json({
+      error: error.code,
+      message: "Couldn't kickout peer from the event. Please try again later",
```
Comment (iamitprakash) [inline controllers/events.js:257]: You can't remove selected Participant from Remove, Please ask Admin or Host for help

File: middlewares/validators/events.js [hunk 3]
```diff
@@ -123,6 +123,27 @@ const addPeerToEvent = async (req, res, next) => {
...
+    reason: joi.string().required(),
+  });
+
+  const validationOptions = { abortEarly: false };
+
+  try {
+    await schema.validateAsync({ id, peerId, reason }, validationOptions);
+    next();
+  } catch (error) {
+    logger.error(`Error while removing a peer from the event: ${error}`);
```
Comment (iamitprakash) [inline middlewares/validators/events.js:142]: We encountered some error while removing selected Participant from event

File: models/events.js [hunk 4]
```diff
@@ -128,9 +128,52 @@ const addPeerToEvent = async (peerData) => {
...
+    if (!peerSnapshot.exists) {
+      throw new Error("Peer not found");
+    }
+
+    const peerData = peerSnapshot.data();
+    const joinedEvents = peerData.joinedEvents;
+
+    const eventIndex = joinedEvents.findIndex((event) => event.event_id === eventId);
+    if (eventIndex === -1) {
+      throw new Error("Peer is not part of the specified event");
```
Comment (iamitprakash) [inline models/events.js:156]: replace Peer with Participants

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: models/events.js (hunk 4 above)
Comment (prerana1821) [reply models/events.js]: Done. Thank you

File: controllers/events.js [hunk 5]
```diff
@@ -215,13 +216,45 @@ const addPeerToEvent = async (req, res) => {
...
+
+/**
+ * Kicks out a peer from an event.
+ *
+ * @async
+ * @function
+ * @param {Object} req - The Express request object.
+ * @param {Object} res - The Express response object.
+ * @returns {Promise<Object>} The JSON response with a success message if the peer is successfully kicked out.
+ * @throws {Object} The JSON response with an error message if an error occurred while kicking out the peer.
```
Comment (vvaibhavdesai) [inline controllers/events.js:238]: where are we throwing error in the below function

File: controllers/events.js [hunk 6]
```diff
@@ -215,13 +216,45 @@ const addPeerToEvent = async (req, res) => {
...
+  const payload = {
+    peer_id: req.body.peerId,
+    reason: req.body.reason,
+  };
+
+  try {
+    await apiService.post(`/active-rooms/${id}/remove-peers`, payload);
+    await eventQuery.kickoutPeer({ eventId: id, peerId: payload.peer_id, reason: req.body.reason });
+    return res.status(200).json({
+      message: `Selected Participant is removed from event.`,
```
Comment (vvaibhavdesai) [inline controllers/events.js:251]: put all the strings in constant

**Review by iamitprakash — APPROVED** [review]

**Pull request merged**